### PR TITLE
fix: revert restriction on psutil and wrapt versions

### LINKED ISSUE #7142: Is it possible to relax the `psutil` dependency in 2.0.0 to prevent dependency version conflicts?
### Which version of dd-trace-py are you using?

2.0.0

### Which version of pip are you using?

Poetry 1.6.1

### Which libraries and their versions are you using?

```
taskipy==1.12.0
```

### How can we reproduce your problem?

Try to install `ddtrace==2.0.0` and `taskipy==1.12.0` in a single Poetry virtual environment

### What is the result that you get?

```
Because ddtrace (2.0.0) depends on psutil (5.6.7)
 and taskipy (1.12.0) depends on psutil (>=5.7.2,<6.0.0), ddtrace (2.0.0) is incompatible with taskipy (1.12.0).
```

### What is the result that you expected?

Installation succeeds, because ddtrace supports a wider range of versions of the psutil dependency:

https://github.com/DataDog/dd-trace-py/blob/e54c6efdc966468d398c1f18fdcef2b09e8ab82c/pyproject.toml#L33

(at least something equal to or newer than `5.7.2`)

### PR DESCRIPTION
**Pull request by emmettbutler**
This pull request fixes #7142 by relaxing the version restriction for the `psutil` and `wrapt` dependencies, which had been pinned to specific versions. Pinning specific versions is problematic in library code.

The `psutil` dependency is upgraded slightly to depend on  the first version where wheels for Linux, Mac and Windows are all available.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -544,7 +544,7 @@ def get_ddup_ext():
         "xmltodict>=0.12",
         "envier",
         "opentelemetry-api>=1",
-        "psutil==5.6.7",
+        "psutil>=5.8.0",
         "setuptools; python_version>='3.12'",
         "wrapt==1.15.0",
```
Comment (sileht) [inline setup.py:549]: You should also remove this one?

**Comment by sileht** [comment]
I was also working on it: https://github.com/DataDog/dd-trace-py/pull/7159
Feel free to close mine.

**Review by mabdinur — DISMISSED** [review]

**Review by P403n1x87 — DISMISSED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -30,12 +30,12 @@ dependencies = [
     "envier",
     "importlib_metadata; python_version<'3.8'",
     "opentelemetry-api>=1",
-    "psutil==5.6.7",
+    "psutil>=5.8.0",
```
Comment (majorgreys) [inline pyproject.toml:33]:
```suggestion
    "psutil~=5.6",
```

**Review by majorgreys — COMMENTED** [review]
Can we use compatible release operators?

Also, I might have missed it, but why the bump for psutil?

**Comment by emmettbutler** [comment]
@majorgreys 5.8.0 is the first version where wheels for Linux, Mac and Windows are all available.

**Review by mabdinur — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**